src/context/volume-context.js

```diff
@@ -0,0 +1,82 @@
+import {createContext, useContext, useReducer} from "react";
+
+const VolumeContext = createContext();
+
+// Custom component to provide the Volume context
+function VolumeProvider(props) {
+  const [state, dispatch] = useReducer(volumeReducer, {
+    toggles: {
+      measurement: 0,
+      season: 0, 
+      tide: 0,
+    },
+    path: "./assets/models/summer_low_salt.png",
+  })
+
+  const value = {state, dispatch}
+  return (
+    <VolumeContext.Provider value={value}>
+      {props.children}
+    </VolumeContext.Provider>
+  );
+}
+
+// Custom hook to get the current VolumeContext 
+function useVolume() {
+  const context = useContext(VolumeContext);
+  if (context === undefined) {
+    throw new Error("useVolume must be used within a VolumeProvider")
+  }
+  return context;
+}
+
+// Custom reducer to update the VolumeContext
+function volumeReducer(state, action) {
+  function getPath() {
+    let path = "";
+    state.toggles.season ? path += "winter_" : path += "summer_";
+    state.toggles.tide ? path += "high_" : path += "low_";
+    state.toggles.measurement ? path += "temp" : path += "salt";
+    path += ".png"
+    
+    console.log("Loading Model:", path)
+    return "./assets/models/" + path;
+  }
+
+  switch(action.type) {
+    case "TOGGLE_MEASUREMENT": {
+      state.toggles = {...state.toggles, measurement: action.payload}
+      state.path = getPath()
+      return state;
+    }
+    case "TOGGLE_SEASON": {
+      state.toggles = {...state.toggles, season: action.payload}
+      state.path = getPath()
+      return state;
+    }
+    case "TOGGLE_TIDE": {
+      state.toggles = {...state.toggles, tide: action.payload}
+      state.path = getPath()
+      return state;
+    }
+    default: {
+      throw new Error(`Unhandled action type: ${action.type}`)
+    }
+  }
+}
```

Comment:
This works great, and you implemented it perfectly. If the state in this system were any smaller, I wouldn't have dealt with a reducer here. I generally like to reserve `useReducer` for more complex system

Reply:
Okay yeah makes sense! I wasn't sure if I needed to use `useReducer` no matter what